File: Add Code Here/C/quicksort.c
#include <stdio.h>

int part(int arr[],int p,int r)
{
    int temp;
    int x=arr[r];
    int i=p-1;
    int j;
    for(j=p;j<=r-1;j++)
    {
        if(arr[j]<=x)
        {
            i++;
            temp=arr[i];
            arr[i]=arr[j];
            arr[j]=temp;
        }
    }
    temp=arr[i+1];
    arr[i+1]=arr[r];
    arr[r]=temp;
    return(i+1);

}
void quicksort(int arr[],int p,int r)
{
    
    if(p<r)
    {
        int q=part(arr,p,r);
        quicksort(arr,p,q-1);
        quicksort(arr,q+1,r);
     
    }
}
void main()
{
    int n;
   printf("enter the no.of elements:");
   scanf("%d",&n);
   int arr[n];
   printf("enter the elements of array:");
   for(int i=0;i<n;i++)
   {
        scanf("%d",&arr[i]);
   }
   quicksort(arr,0,n-1);
   printf("sorted array is:");
   for(int i=0;i<n;i++)
   {
       printf("%d",arr[i]);
       printf(" ");
   }
}

File: Add Code Here/GO/quickSort.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func main() {
	rand.Seed(time.Now().UnixNano())
	data := rand.Perm(10) // Generate a random permutation of 10 numbers
	fmt.Println("Unsorted:", data)
	quickSort(data, 0, len(data)-1)
	fmt.Println("Sorted:", data)
}

func quickSort(arr []int, low, high int) {
	if low < high {
		pivot := partition(arr, low, high)
		quickSort(arr, low, pivot-1)
		quickSort(arr, pivot+1, high)
	}
}

func partition(arr []int, low, high int) int {
	pivot := arr[high]
	i := low - 1
	for j := low; j <= high-1; j++ {
		if arr[j] < pivot {
			i++
			arr[i], arr[j] = arr[j], arr[i]
		}
	}
	arr[i+1], arr[high] = arr[high], arr[i+1]
	return i + 1
}
